### PR TITLE
perf: Use selectors on prediction hooks

### PR DESCRIPTION
**Pull request by memoyil**

### CODE REVIEW

File: src/state/predictions/hooks.ts [hunk 1]
```diff
@@ -1,62 +1,47 @@
...
-  return Object.keys(rounds).reduce((accum, epoch) => {
-    return {
-      ...accum,
-      [epoch]: parseBigNumberObj<ReduxNodeRound, NodeRound>(rounds[epoch]),
-    }
-  }, {}) as { [key: string]: NodeRound }
+  return useSelector(getRoundsSelector)
 }
 
 export const useGetRoundsByCloseOracleId = () => {
```
[the remark this answers is not in the document]
Comment (memoyil) [reply src/state/predictions/hooks.ts:24]: Sure, I will do it in separate pr

**Comment by chef-jojo** [comment]
is this something we can measure? 
Honestly I don't like to memo select every redux stuff, just like react memo doesn't by default with reason

**Comment by memoyil** [comment]
> is this something we can measure? Honestly I don't like to memo select every redux stuff, just like react memo doesn't by default with reason

It reduces object creation, order and parseBigNumberObj operations on each render (which happens frequently). In high end devices the diff can be negligible but low end mobile devices those operations can be cumbersome and reducing them can affect performance positively

File: src/state/predictions/selectors.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+export const getRoundsSelector = createSelector([selectRounds], (rounds) => {
+  return Object.keys(rounds).reduce((accum, epoch) => {
+    return {
+      ...accum,
+      [epoch]: parseBigNumberObj<ReduxNodeRound, NodeRound>(rounds[epoch]),
+    }
+  }, {}) as { [key: string]: NodeRound }
+})
+
+export const getSortedRoundsSelector = createSelector([selectRounds], (rounds) => {
```
[the remark this answers is not in the document]
Comment (memoyil) [reply src/state/predictions/selectors.ts]: Done

File: src/state/predictions/selectors.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  return Object.keys(rounds).reduce((accum, epoch) => {
+    const parsed = parseBigNumberObj<ReduxNodeRound, NodeRound>(rounds[epoch])
+    return {
+      ...accum,
+      [parsed.closeOracleId]: parsed,
+    }
+  }, {}) as { [key: string]: NodeRound }
+})
+
+export const getRoundsSelector = createSelector([selectRounds], (rounds) => {
```
[the remark this answers is not in the document]
Comment (memoyil) [reply src/state/predictions/selectors.ts]: Done

File: src/state/predictions/selectors.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+          [epoch]: parseBigNumberObj<ReduxNodeRound, NodeRound>(rounds[epoch]),
+        }
+      }, {}) as { [key: string]: NodeRound },
+    ),
+    ['epoch'],
+    ['asc'],
+  )
+})
+
+export const getCurrentRoundSelector = createSelector([selectCurrentEpoch, selectRounds], (currentEpoch, rounds) => {
```
[the remark this answers is not in the document]
Comment (memoyil) [reply src/state/predictions/selectors.ts]: Done

File: src/state/predictions/selectors.ts [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    return {
+      ...accum,
+      [epoch]: parseBigNumberObj<ReduxNodeRound, NodeRound>(rounds[epoch]),
+    }
+  }, {}) as { [key: string]: NodeRound }
+
+  return roundData[currentEpoch]
+})
+
+export const getMinBetAmountSelector = createSelector([selectMinBetAmount], (minBetAmount) => {
```
[the remark this answers is not in the document]
Comment (memoyil) [reply src/state/predictions/selectors.ts]: Done

**Pull request merged**